### ISSUE TITLE
Dynamically loading on Mac OS X causes problems without trampoline

### ISSUE THREAD

**Issue by dyoo** [pb82/sdl4racket#1]
When using the bindings on Mac OS X, the user will see an exception on screen instantiation unless the program's running under a Cocoa context.  I've written a module to do the necessary work:

```
https://gist.github.com/4526379
```

(Comment the use of initialize-ns-app in that code.)

initialize-ns-app provides a Cocoa context in which all the rest of SDL usage must occur on Mac OS X.

If you don't do this, you'll see this exception as soon as you call sdl-set-video-mode:

2013-01-13 15:07:22.716 racket[26263:707] **\* Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Error (1000) creating CGSWindow on line 259'
**\* First throw call stack:
(
    0   CoreFoundation                      0x00007fff8e6ca0a6 __exceptionPreprocess + 198
    1   libobjc.A.dylib                     0x00007fff9681e3f0 objc_exception_throw + 43
    2   CoreFoundation                      0x00007fff8e6c9e7c +[NSException raise:format:] + 204
    3   AppKit                              0x00007fff951b2c29 _NSCreateWindowWithOpaqueShape2 + 655
    4   AppKit                              0x00007fff951b1420 -[NSWindow _commonAwake] + 2002
    5   AppKit                              0x00007fff9516fe62 -[NSWindow _commonInitFrame:styleMask:backing:defer:] + 1763
    6   AppKit                              0x00007fff9516efaf -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1568
    7   AppKit                              0x00007fff9516e97f -[NSWindow initWithContentRect:styleMask:backing:defer:] + 45
    8   SDL                                 0x0000000102f515cc SDL_SoftStretch + 24672
    9   SDL                                 0x0000000102f4fe38 SDL_SoftStretch + 18636
    10  SDL                                 0x0000000102f44da1 SDL_SetVideoMode + 892
    11  Racket                              0x00000001002be82c ffi_call_unix64 + 76
    12  ???                                 0x00007fff5fbfd8d0 0x0 + 140734799796432
)
libc++abi.dylib: terminate called throwing an exception
Abort trap: 6

**Comment by pb82** [comment]
Great, thanks for your Effort! Since you already solved that in your fork, would you mind sending me a pull request?

**Comment by dyoo** [comment]
On Tue, Jan 15, 2013 at 1:18 PM, Peter Braun notifications@github.comwrote:

> Great, thanks for your Effort! Since you already solved that in your fork,
> would you mind sending me a pull request?

Will do after I clean it up a little.  The other thing I need to check is
what happens under DrRacket.

**Comment by dyoo** [comment]
Ok, so I've investigated the issue more closely, and it looks like just loading racket/gui/base is enough to get the right context, so I've removed my trampoline and replaced it with a plain require.  :)

My branch includes a prepackaged lib that holds the libraries for all the major platforms.  I haven't carefully added a LICENCE file to say where each binary came from yet, so that something more we'd have to do to be perfectly good with the licensing, right?  I'll send that pull request.
